ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,76 @@
+import axios from 'axios'
+import { randomUUID } from 'crypto'
+import FormData from 'form-data'
+import { lastValueFrom } from 'rxjs'
+import { HttpService } from '@nestjs/axios'
+import { ConfigService } from '@nestjs/config'
+import { Injectable } from '@nestjs/common'
+
+import { apiURL, CirrusAPIEndPoint, cirrusAPIVersion, CirrusErrorData, CirrusInvitationResponse } from './cirrus-invitation.interfaces'
+import { InvitationAPIError } from './invitation.errors'
+
+import { Config } from '../config'
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor(
+    private readonly configService: ConfigService<Config, true>,
+    private readonly httpService: HttpService
+  ) {
+    this.url = apiURL + cirrusAPIVersion + CirrusAPIEndPoint
```

Comment:
This is throwing a typescript linting errors because of concatenation. This needs to modified in order to eliminate the error.

Reply:
@pushyamig, I'd appreciate advice on getting an accurate report of eslint errors in the code.

Do you view eslint error reports mainly via Codacy, in your IDE, or some other way, like CLI?

When I use IntelliJ IDEA, I don't have eslint configured well in it, so a lot of errors slip past me.  I don't know about the errors until I see them in the Codacy report later, which is delayed.  Then I end up in a fix/commit/wait loop trying to resolve the errors.

I hoped when I started trying VSC the eslint problem would be much improved.  However, it seems that eslint problems only appear in VSC's "Problems" view when each file is open.  If a file containing an eslint error isn't open in an editor, the error isn't shown in "Problems".

I wonder whether you (or any of @ssciolla, @jonespm, and @zqian) have any better experience of viewing eslint error reports locally, before committing to GitHub and waiting for Codacy.

Anyway, rest assured that as long as GitHub continues to show that the PR's tests have failed due to Codacy errors, I'm aware of the problems and I'm working on them.  They get lower priority than fixing functional code problems, however.

Reply:
Just to summarize…

@pushyamig invited me to a Zoom meeting where she showed me how she views eslint error reports.  It's basically the same thing that I had been doing in VSC.  It's awkward.

Additionally, eslint in Codacy reports more/different errors than it does in VSC.  Pushyami told me that @ssciolla has been concerned about the eslint errors in VSC only, ignoring those via Codacy.  I think there's probably more to it than that, because that would mean the Codacy check in GH would never pass and our PRs wouldn't be mergeable.  (At least not by a non-admin.)

I think we need to discuss in future dev. team meetings…

* Configuring eslint for projects to produce the same results in Codacy, IDE(s), and CLI
* How to run eslint in those various environments and document it

Reply:
Hi @lsloan,

Here's more background:

1) There is a known issue with Codacy determining proper types of variables derived from libraries. There are ways to get around this by setting up a custom GitHub action, but it has not been a top priority for me. This is documented in https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/215.

2) I have previously assisted you (over Slack) in on how to run `eslint` via the command line. If you aren't satisfied with VSCode's problem reporting (saw some improvement ideas [here](https://stackoverflow.com/questions/41702815/how-to-make-visual-studio-code-check-entire-project-for-errors)), you can run `eslint` directly like so:
```
// If you don't yet have it installed
npm install -g eslint
// from application root, for the backend
eslint ccm_web/server/src 
```

Reply:
Yes, we have met about it.  I remember that and I appreciated the discussion.

I'm bothered that the IDE/CLI results don't match up with Codacy's.  I wasn't aware of a bug in Codacy.  I only saw that it appears to be running the same version of eslint that we do locally, so I don't understand why theirs doesn't work correctly.